drivers/sensor/nuvoton_adc_cmp_npcx/adc_cmp_npcx.c

```diff
@@ -0,0 +1,237 @@
+/*
+ * Copyright (c) 2022 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <drivers/sensor.h>
+#include <drivers/adc/adc_npcx_threshold.h>
+#include <drivers/sensor/adc_cmp_npcx.h>
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(adc_cmp_npcx, LOG_LEVEL_ERR);
+
+#define ADC_CMP_NPCX_UNDEFINED		(-1)
+
+struct adc_cmp_npcx_data {
+	/* Work queue to be notified when threshold assertion happens */
+	struct k_work work;
+	/* Sensor trigger hanlder to notify user of assetion */
+	sensor_trigger_handler_t handler;
+	/* ADC NPCX driver reference */
+	const struct device *dev;
+};
+
+struct adc_cmp_npcx_config {
+	/*
+	 * Pointer of ADC device that will be performing measurement, this
+	 * must be provied by device tree.
+	 */
+	const struct device *adc;
+	/*
+	 * ADC channel that will be used to measure signal, this must be
+	 * provided by device tree.
+	 */
+	uint8_t chnsel;
+	/* Threshold selection number */
+	uint8_t th_sel;
+	/* Threshold assertion value */
+	uint16_t thrval;
+	/* Compariosn between measured value and threshold */
+	enum adc_cmp_npcx_comparison comparison;
+};
+
+#define DT_DRV_COMPAT nuvoton_adc_cmp
+
+#define DRV_CONFIG(dev) ((const struct adc_cmp_npcx_config *)(dev)->config)
+#define DRV_DATA(dev) ((struct adc_cmp_npcx_data *)(dev)->data)
```

Comment:
In general, new drivers should use this style:

```
const struct foo_config *config = dev->config;
````

inside functions instead of defining these.  See https://github.com/zephyrproject-rtos/zephyr/pull/41674#pullrequestreview-853663582 for more information.

Reply:
ACK